src/main/java/uk/gov/pay/api/filter/ratelimit/RedisRateLimiter.java

```diff
@@ -52,20 +53,14 @@ synchronized void checkRateOf(String accountId, RateLimiterKey key)
 
     synchronized private Long updateAllowance(String key) {
         String derivedKey = getKeyForWindow(key);
+        StatefulRedisConnection<String, String> connection = redisClient.connect();
```

Comment:
Do we need to close this connection or does lettuce takes care of it ? Java docs says `connect()` opens new connection to Redis https://lettuce.io/core/release/api/io/lettuce/core/RedisClient.html#connect-- 

Also with Jedis, we get connection pool. Is this something that will be useful ?

Reply:
Thanks for catching this out. I think the `StatefulRedisConnection` can be a field in the class. However, given that `updateAllowance` is a synchronized method, I don't think there's any value in using a connection pool here.

Reply:
@kbottla I've closed the connection via the (renamed) RedisClientManager now.